internal/watchdhelpers/watchdhelpers.go

```diff
@@ -0,0 +1,102 @@
+package watchdhelpers
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ubuntu/adsys/internal/i18n"
+	"gopkg.in/yaml.v2"
+)
+
+// AppConfig represents the configurable options of the application.
+type AppConfig struct {
+	Verbose int
+	Force   bool
+	Dirs    []string
+}
+
+// GetDirsFromConfigFile unmarshals and returns the directories from the passed in
+// config file.
+func GetDirsFromConfigFile(configFile string) []string {
+	var dirs []string
+	config, err := os.ReadFile(configFile)
+	if err != nil {
+		return dirs
+	}
+	cfg := AppConfig{}
+	if err := yaml.Unmarshal(config, &cfg); err == nil {
+		dirs = cfg.Dirs
+	}
+	return dirs
+}
+
+// FilterAbsentDirs returns only the existing directories from the passed in
+// slice.
+func FilterAbsentDirs(dirs []string) []string {
+	var filtered []string
+	for _, dir := range dirs {
+		if stat, err := os.Stat(dir); err == nil && stat.IsDir() {
+			filtered = append(filtered, dir)
+		}
+	}
+	return filtered
+}
+
+// WriteConfig writes the config to the given file, checking whether the
+// directories that are passed in actually exist. It receives a config file and
+// a slice of absolute sorted paths.
+func WriteConfig(confFile string, dirs []string, defaultConfFile string) error {
+	if len(dirs) == 0 {
+		return fmt.Errorf(i18n.G("needs at least one directory to watch"))
+	}
+
+	// Make sure all directories exist
+	for _, dir := range dirs {
+		if _, err := os.Stat(dir); errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf(i18n.G("directory %q does not exist"), dir)
+		}
+	}
+
+	// Empty input means using the default config file
+	if confFile == "" {
+		confFile = defaultConfFile
```

Comment:
I would simplify here and do that on the caller side. Basically the API is "write it there"